h5py/h5t.pyx

```diff
@@ -946,16 +946,29 @@ cdef class TypeFloatID(TypeAtomicID):
         size = self.get_size()                  # int giving number of bytes
         order = _order_map[self.get_order()]    # string with '<' or '>'
 
-        if size == 2 and not hasattr(np, 'float16'):
+        s_offset, e_offset, e_size, m_offset, m_size = self.get_fields()
+        e_bias = self.get_ebias()
+
+        # Handle non-standard exponent and mantissa sizes.
+        if m_size > 112 or (2**e_size - 2 - e_bias) > 16383 or (1 - e_bias) < -16382:
+            raise ValueError('Invalid exponent or mantissa size in ' + str(self))
+        elif m_size > 52 or (2**e_size - 2 - e_bias) > 1023 or (1 - e_bias) < -1022:
```

Comment:
Yes, that's what the last of the four tests does check, the float on 2 bytes gets promoted twice up to a float 64 (8 bytes)